docs/dev/testing/test-fixtures.md

```diff
@@ -1,9 +1,357 @@
+
+### Cleanup Strategies
+
+**Strategy 1: RAII Guards (Preferred)**
+- Automatic cleanup on test completion or failure
+- Used by both SDK and CLI tests
+- Most reliable for preventing orphaned resources
+
+**Strategy 2: Periodic Cron Job**
+- GitHub Actions workflow runs periodically
+- Cleans up deployments older than 4 hours
+- Captures deployments from both `pr-integration-test-*` and `release-integration-test-*` patterns
+- Backup cleanup mechanism for orphaned deployments
+
+**Strategy 3: Manual Cleanup**
+- Use LangSmith UI to bulk delete old test deployments
+- Use CLI commands (once implemented):
+  ```bash
+  # List test deployments
+  langstar graph list | grep "integration-test"
+
+  # Delete specific deployment
+  langstar graph delete <deployment-id>
+  ```
+
+## Shared vs Fresh Deployments
+
+### Shared Deployments (SDK)
+
+**Pattern:** `pr-integration-test-{timestamp}`
+
+**Advantages:**
+- Faster test execution (~6 min vs ~22 min)
+- Reduced API calls and resource usage
+- Better for development iteration
+
+**Disadvantages:**
+- Tests must not modify deployment state
+- Requires coordination between tests
+- May hide deployment lifecycle bugs
+
+**Use for:**
+- SDK PromptHub tests
+- SDK deployment workflow tests (default path)
+- Read-only operations
+
+### Fresh Deployments (CLI, SDK Release Tests)
+
+**Pattern:** `release-integration-test-{timestamp}`
+
+**Advantages:**
+- Complete lifecycle validation
+- No test interdependencies
+- Tests deployment cleanup functionality
+- Catches deployment-specific bugs
+
+**Disadvantages:**
+- Slower (creates new deployment each run)
+- More API calls and resource usage
+- Longer test execution time
+
+**Use for:**
+- CLI graph command tests
+- SDK deployment workflow full lifecycle tests
+- Pre-release validation
+- CRUD operations that modify state
+
+## Test Isolation
+
+### Unique Naming for Parallel Safety
+
+Test resources use microsecond timestamps + UUID suffix for uniqueness:
+
+```rust
+fn generate_test_name(prefix: &str) -> String {
+    let timestamp = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .as_micros();
+    let uuid_suffix = &Uuid::new_v4().to_string()[..8];
+    format!("{}-{}-{}", prefix, timestamp, uuid_suffix)
+}
+```
+
+This prevents name collisions even when tests run concurrently.
+
+### Selective Serialization
+
+**Most tests run in parallel** by default for better performance.
+
+**Tests with shared resources** are marked with `#[serial]`:
+- CLI assistant tests that share a deployment via `OnceLock`
+- Tests that modify global state
+- Tests with resource conflicts
+
+Use `#[serial]` only when truly necessary - prefer parallel execution for faster CI.
+
+## Maintenance
+
+### Updating the Test Graph
+
+To update the graph logic:
+
+1. Modify `tests/fixtures/test-graph-deployment/test_agent/agent.py`
+2. Commit changes
+3. Redeploy via LangSmith UI or CI/CD
+4. Update `TEST_GRAPH_ID` if deployment changes
```

Comment:
review for factuality

Reply:
Fixed in commit 94e7b90: Removed step 4 "Update TEST_GRAPH_ID if deployment changes" from the update procedure. Tests now auto-discover deployments.